solver/src/driver.rs

```diff
@@ -346,43 +329,36 @@ impl Driver {
                 .get(&self.native_token)
                 .expect("Price of native token must be known.");
 
-        futures::stream::iter(settlements)
-            .filter_map(|(solver, settlement)| async {
-                let surplus = settlement.total_surplus(prices);
-                let scaled_solver_fees = settlement.total_scaled_unsubsidized_fees(prices);
-                let gas_estimate = match settlement_simulation::simulate_and_estimate_gas_at_current_block(
-                    std::iter::once((solver.account().clone(), settlement.clone())),
-                    &self.settlement_contract,
-                    &self.web3,gas_price
-                )
-                .await
-                .map(|results|  results.into_iter().next().unwrap().map_err(Error::from))
-                {
-                    Ok(Ok(gas)) => gas,
-                    Err(err)  | Ok(Err(err)) => {
-                        tracing::error!("Failed to estimate gas for solver {}: {:?}", solver.name(), err);
-                        return None;
-                    }
-                };
-                let rated_settlement = RatedSettlement {
-                    settlement,
-                    surplus,
-                    solver_fees: scaled_solver_fees,
-                    gas_estimate,
-                    gas_price: gas_price_normalized.clone(),
-                };
-                tracing::info!(
-                    "Objective value for solver {} is {}: surplus={}, gas_estimate={}, gas_price={}",
-                    solver.name(),
-                    rated_settlement.objective_value(),
-                    rated_settlement.surplus,
-                    rated_settlement.gas_estimate,
-                    rated_settlement.gas_price,
-                );
-                Some((solver, rated_settlement))
-            })
-            .collect::<Vec<_>>()
-            .await
+        let rate_settlement = |solver, settlement: Settlement, gas_estimate| {
```

Comment:
Sort of a general note - I quite like these closures that you use to organize logic to keep the `partition_map` closure leaner and easier to follow!